lib/puppet/indirector/generic_http.rb

```diff
@@ -0,0 +1,38 @@
+require 'puppet/file_serving/terminus_helper'
+require 'net/http'
+require 'puppet/util/http_proxy'
+
+class Puppet::Indirector::GenericHttp < Puppet::Indirector::Terminus
+  desc "Retrieve data from a remote HTTP server."
+
+  include Puppet::FileServing::TerminusHelper
+
+  class <<self
+    attr_accessor :http_method
+  end
+
+  def find(request)
+    uri = URI( unescape_url(request.key) )
+
+    proxy_class = Net::HTTP::Proxy(Puppet::Util::HttpProxy.http_proxy_host,
```

Comment:
Oh, I guess this is because we don't want to use Puppet's certs to talk to the other web server?